team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java

```diff
@@ -1,55 +1,86 @@
 import ua.knu.knudev.teammanagerapi.api.AccountProfileApi;
-import ua.knu.knudev.knudevcommon.utils.AcademicUnitsIds;
 import ua.knu.knudev.teammanagerapi.dto.AccountProfileDto;
 import ua.knu.knudev.teammanagerapi.exception.AccountException;
 import ua.knu.knudev.teammanagerapi.response.AccountRegistrationResponse;
 
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+
 @Service
 @RequiredArgsConstructor
+@Validated
+@Slf4j
 public class AccountProfileService implements AccountProfileApi {
 
     private final AccountProfileRepository accountProfileRepository;
-    private final AccountProfileMapper accountProfileMapper;
     private final AccountAuthServiceApi accountAuthServiceApi;
-    private final FileServiceApi fileServiceApi;
+    private final ImageServiceApi imageServiceApi;
     private final DepartmentService departmentService;
 
     @Override
     @Transactional
-    public AccountRegistrationResponse register(AccountCreationRequest request) {
+    public AccountRegistrationResponse register(@Valid AccountCreationRequest request) {
         validateEmailNotExists(request.email());
         departmentService.validateAcademicUnitByIds(request.academicUnitsIds());
 
         AuthAccountCreationResponse createdAuthAccount = accountAuthServiceApi.createAccount(request);
-        String uploadFilename = fileServiceApi.uploadAccountPicture(request.avatarFile());
+        String uploadFilename = uploadAvatar(request.avatarFile());
 
         AccountProfile accountProfileToSave = buildAccountProfile(request, uploadFilename, createdAuthAccount);
         AccountProfile savedAccount = accountProfileRepository.save(accountProfileToSave);
 
-        return buildRegistrationResponse(savedAccount, request.email());
+        return buildRegistrationResponse(savedAccount, request.email(), createdAuthAccount);
+    }
+
+    private String uploadAvatar(MultipartFile file) {
+        try {
+            boolean fileIsPresent = ObjectUtils.isNotEmpty(file) && ArrayUtils.getLength(file.getBytes()) != 0;
+            if (fileIsPresent) {
+                return imageServiceApi.uploadFile(file, ImageSubfolder.ACCOUNT_PICTURES);
+            }
+        } catch (IOException e) {
+            throw new FileException("Error while reading the file.");
+        }
+        return null;
     }
 
     private void validateEmailNotExists(String email) {
-        if (accountProfileRepository.existsByEmail(email)) {
+        boolean accountProfileExists = accountProfileRepository.existsByEmail(email);
+        if (accountProfileExists) {
             throw new AccountException(
                     String.format("Account with email %s already exists", email)
             );
         }
+
+        boolean accountAuthExists = accountAuthServiceApi.existsByEmail(email);
+        if (accountAuthExists) {
+            log.error("AccountProfile does not exists, but AccountAuth does exist by email {}", email);
```

Comment:
You will throw exception with message already, so you don't need message

Reply:
This is very rare case. It ideally would never happen, but if it does - we will see the logs